libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -44,7 +49,7 @@ internal class MapboxDirectionsSession(
         @RoutesExtra.RoutesUpdateReason routesUpdateReason: String
     ) {
         this.initialLegIndex = initialLegIndex
-        if (this.routes.isEmpty() && routes.isEmpty()) {
+        if (routesInitialized && this.routes.isEmpty() && routes.isEmpty()) {
             return
         }
```

Comment:
should be? otherwise, `routeInitialized` is always `false` and `setRoutes` is never success 
```suggestion
        if (this.routes.isEmpty() && routes.isEmpty()) {
            routesInitialized = true
            return
        }
```

Reply:
No, it's OK. The point is that `this.routes.isEmpty() && routes.isEmpty()` can evaluate to true if we set empty routes without previously setting non-empty ones. I think that's explicit action which should be considered. So if routes were not initialised we set them no matter if they're empty or not. 
If it's false we just skip the return and set the routes, right? And then it is set to true. When it's true, we ignore setting empty routes over empty routes.